frontend/src/store/useTaskStore.js

```
@@ -0,0 +1,103 @@
+  updateTask: async (id, updatedData) => {
+    set({ isUpdatingTask: true, taskError: null });
+    try {
+      const response = await axiosInstance.put(`/tasks/${id}`, updatedData);
+      set((state) => ({
+        tasks: state.tasks.map((task) =>
+          task._id === id ? response.data : task
+      ),
+    }));
+  } catch (error) {
+    console.error("Error updating task:", error);
+    set({ taskError: "Failed to update task: " + error.message });
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)